compiler/src/dmd/expressionsem.d

```diff
@@ -6436,7 +6436,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 return setError();
 
             checkFunctionAttributes(exp, sc, exp.f);
-            checkAccess(exp.loc, sc, null, exp.f);
+            checkAccess(isSuper ? cd.baseClass : ad, exp.loc, sc, exp.f);
```

Comment:
Note that `checkAccess` first gets the most visible overload of `exp.f` and then performs access checks, so if you had a public constructor in Base you would still end up incorrectly calling the private constructor. This is (Issue 3254)[https://issues.dlang.org/show_bug.cgi?id=3254] all over again. However, in this case, I think we can fix it by calling `checkSymbolAccess` which checks the symbol directly.

(This will not break code, because it technically impossible to pass super as a template alias parameter. for more info please see: https://github.com/dlang/dmd/pull/13257 )